### PR TITLE
Example to dump a dataframe in CSV format to a S3-compatible object store

### PR DESCRIPTION
**Pull request by Qqwy**
This PR is a draft on how to dump a dataframe to a S3-compatible object store.
In this particular example, the CSV format is used; using JSON, Partque or IPC format should only require minimal changes.
Also, to make trying out easy, it currently uses the hard-coded `object_store::LocalFileSystem` as object store.

I've used the following to test the functionality manually for now:

```elixir
iex> mountains = Explorer.DataFrame.new(name: ["Everest", "K2", "Aconcagua"], elevation: [8848, 8611, 6962])
iex> Explorer.PolarsBackend.DataFrame.to_csv_writer_sample(mountains, "/tmp/mountains.csv", false, ",")
iex> File.read!("/tmp/mountains.csv")
"Everest,8848\nK2,8611\nAconcagua,6962\n"
```

Once properly integrated (main blocker: how to set up and pass in the configuration of the object store?), this could be turned into a full test.

### CODE REVIEW

**Comment by philss** [comment]
hey @Qqwy, thanks for this PR!

I was about to start working on features based on it, but after rebasing, I got a clippy issue that I don't know how to fix :grimacing: 
Do you mind to take a look? Thanks!

**Comment by Qqwy** [comment]
@philss done! It was not actually necessary to wrap `ObjectStore` in an `Arc<Mutex<...>>` because the trait itself requires implementations to manage thread-safety internally anyway (it comes with a `Sync` bound), so I could remove the manual wrapper 😎 .

File: lib/explorer/polars_backend/data_frame.ex [hunk 1]
```diff
@@ -129,6 +129,15 @@ defmodule Explorer.PolarsBackend.DataFrame do
     end
   end
 
+  def to_csv_writer_sample(%DataFrame{data: df}, path, header?, delimiter) do
```
Comment (philss) [inline lib/explorer/polars_backend/data_frame.ex:132]: Please just move this function to after the next clause of `to_csv`. I think this is the reason the tests are breaking.

**Comment by philss** [comment]
I'm planning to merge this PR and work upon it. What do you think, @Qqwy and @josevalim ?

**Comment by josevalim** [comment]
@philss yes, please go ahead. This is meant to be a proof of concept after all! :)

**Comment by philss** [comment]
@Qqwy thank you! :purple_heart:

**Pull request merged**